MathlibTest/superscript.lean

```diff
@@ -81,3 +81,150 @@ run_cmd Lean.Elab.Command.liftTermElabM do
   Lean.logInfo <| ← Lean.PrettyPrinter.ppTerm <| ← `(term| ($a)$α:subscript)
 
 end
+
+section delab
+
+open Lean PrettyPrinter.Delaborator SubExpr
+open Mathlib.Tactic (delabSubscript delabSuperscript)
+
+private def check_subscript {α : Type} (_ : α) := ()
+local syntax:arg "test(" noWs subscript(term) noWs ")" : term
+local macro_rules | `(test($a:subscript)) => `(check_subscript $a)
+
+private def check_superscript {α : Type} (_ : α) := ()
+local syntax:arg "test(" noWs superscript(term) noWs ")" : term
+local macro_rules | `(test($a:superscript)) => `(check_superscript $a)
+
+@[app_delab check_subscript]
+private def delabCheckSubscript : Delab := withOverApp 2 do
+  let sub ← withAppArg delabSubscript
+  `(test($sub:subscript))
+
+@[app_delab check_superscript]
+private def delabCheckSuperscript : Delab := withOverApp 2 do
+  let sup ← withAppArg delabSuperscript
+  `(test($sup:superscript))
+
+universe u v w
+
+/-- `α` can not be subscripted or superscripted. -/
+private def α {γ : Type u} {δ : Type v} : γ → δ → δ := fun _ ↦ id
+/-- `β` can be both subscripted and superscripted. -/
+private def β {γ : Type u} {δ : Type v} : γ → δ → δ := fun _ ↦ id
+
+/-- `d` can not be subscripted, so we create an alias for `id`. -/
+private abbrev ID {γ : Sort u} := @id γ
+
+variable (n : String)
+
+section subscript
+
+/-- info: test(₁₂₃₄₅₆₇₈₉₀ ₌₌ ₁₂₃₄₅₆₇₈₉₀) : Unit -/
+#guard_msgs in #check test(₁₂₃₄₅₆₇₈₉₀ ₌₌ ₁₂₃₄₅₆₇₈₉₀)
+/-- info: test(ᵦ ₙ ₍₁ ₊ ₂ ₋ ₃ ₌ ₀₎) : Unit -/
+#guard_msgs in #check test(ᵦ ₙ ₍₁ ₊ ₂ ₋ ₃ ₌ ₀₎)
+/-- info: test(ᵦ) : Unit -/
+#guard_msgs in #check test(ᵦ)
+/-- info: test(ᵦ ₍₎) : Unit -/
+#guard_msgs in #check test(ᵦ ₍₎)
+/-- info: test(ᵦ ᵦ ᵦ ᵦ) : Unit -/
+#guard_msgs in #check test(ᵦ ᵦ ᵦ ᵦ)
+/-- info: test(ɪᴅ ɪᴅ ₃₇) : Unit -/
+#guard_msgs in #check test(ɪᴅ ɪᴅ ₃₇)
+
+end subscript
+
+section superscript
+
+/-- info: test(¹²³⁴⁵⁶⁷⁸⁹⁰ ⁼⁼ ¹²³⁴⁵⁶⁷⁸⁹⁰) : Unit -/
+#guard_msgs in #check test(¹²³⁴⁵⁶⁷⁸⁹⁰ ⁼⁼ ¹²³⁴⁵⁶⁷⁸⁹⁰)
+/-- info: test(ᵝ ⁿ ⁽¹ ⁺ ² ⁻ ³ ⁼ ⁰⁾) : Unit -/
+#guard_msgs in #check test(ᵝ ⁿ ⁽¹ ⁺ ² ⁻ ³ ⁼ ⁰⁾)
+/-- info: test(ᵝ) : Unit -/
+#guard_msgs in #check test(ᵝ)
+/-- info: test(ᵝ ⁽⁾) : Unit -/
+#guard_msgs in #check test(ᵝ ⁽⁾)
+/-- info: test(ᵝ ᵝ ᵝ ᵝ) : Unit -/
+#guard_msgs in #check test(ᵝ ᵝ ᵝ ᵝ)
+/-- info: test(ⁱᵈ ⁱᵈ ³⁷) : Unit -/
+#guard_msgs in #check test(ⁱᵈ ⁱᵈ ³⁷)
+
+end superscript
+
+variable (x_x : String)
+
+private def card {γ : Sort u} := @id γ
+local prefix:arg "#" => card
+
+private def factorial {γ : Sort u} := @id γ
+local notation:10000 n "!" => factorial n
+
+private class HSMul (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  hSMul : α → β → γ
+instance : HSMul Nat Nat Nat where
+  hSMul a b := a * b
+infixr:73 " • " => HSMul.hSMul
```

Comment:
If this isn't imported I think you can drop this test; the `^` one is sufficient, especially given your new implementation.